optimizely/config/polling_manager.go

```diff
@@ -83,22 +93,38 @@ func (cm *PollingProjectConfigManager) activate(initialPayload []byte, init bool
 	}
 }
 
-// NewPollingProjectConfigManager returns new instance of PollingProjectConfigManager
-func NewPollingProjectConfigManager(ctx context.Context, requester *Requester, initialPayload []byte, pollingWait time.Duration) *PollingProjectConfigManager {
+// NewPollingProjectConfigManagerWithOptions returns new instance of PollingProjectConfigManager with the given options
+func NewPollingProjectConfigManagerWithOptions(ctx context.Context, sdkKey string, options PollingProjectConfigManagerOptions) *PollingProjectConfigManager {
 
-	if pollingWait == 0 {
-		pollingWait = defaultPollingWait
+	var requester Requester
+	if options.Requester != nil {
+		requester = options.Requester
+	} else {
+		url := fmt.Sprintf(DatafileURLTemplate, sdkKey)
```

Comment:
are we sure at this point sdkKey cannot be empty?

Reply:
discussed offline, the check for the SDK key happens upstream so we're comfortable not doing it here